carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/FileManagementUtil.java

```diff
@@ -0,0 +1,291 @@
+     */
+    public static void zipFolder(String srcFolder, String destZipFile, Log log) {
+        try (FileOutputStream fileWriter = new FileOutputStream(destZipFile);
+             ZipOutputStream zip = new ZipOutputStream(fileWriter)) {
+            addFolderContentsToZip(srcFolder, zip, log);
+        } catch (IOException e) {
+            log.warn(e);
+        }
+
+
+    }
+
+    private static void addToZip(String path, String srcFile, ZipOutputStream zip, Log log) {
+        File folder = new File(srcFile);
+        if (folder.isDirectory()) {
+            addFolderToZip(path, srcFile, zip, log);
+        } else {
+            // Transfer bytes from in to out
+            byte[] buf = new byte[1024];
+            int len;
+            try (FileInputStream inputStream = new FileInputStream(srcFile)) {
+
+                if (path.trim().equals("")) {
+                    zip.putNextEntry(new ZipEntry(folder.getName()));
+                } else {
+                    zip.putNextEntry(new ZipEntry(path + "/" + folder.getName()));
+                }
+                while ((len = inputStream.read(buf)) > 0) {
+                    zip.write(buf, 0, len);
+                }
+                inputStream.close();
+            } catch (IOException e) {
+                log.warn(e);
+            }
+        }
+    }
+
+    private static void addFolderContentsToZip(String srcFolder, ZipOutputStream zip, Log log) {
+        File folder = new File(srcFolder);
+        String fileList[] = folder.list();
+        if (fileList != null) {
+            try {
+                int i = 0;
+                while (true) {
+                    if (fileList.length == i) {
+                        break;
+                    }
+                    if (new File(folder, fileList[i]).isDirectory()) {
+                        zip.putNextEntry(new ZipEntry(fileList[i] + "/"));
+                        zip.closeEntry();
+                    }
+                    addToZip("", srcFolder + "/" + fileList[i], zip, log);
+                    i++;
+                }
+            } catch (IOException e) {
+                log.warn(e);
+            }
+        }
+    }
+
+    private static void addFolderToZip(String path, String srcFolder, ZipOutputStream zip, Log log) {
+        File folder = new File(srcFolder);
+        String fileList[] = folder.list();
+        if (fileList != null) {
+            try {
+                int i = 0;
+                while (true) {
+                    if (fileList.length == i) {
+                        break;
+                    }
+                    String newPath = folder.getName();
+                    if (!path.equalsIgnoreCase("")) {
+                        newPath = path + "/" + newPath;
+                    }
+                    if (new File(folder, fileList[i]).isDirectory()) {
+                        zip.putNextEntry(new ZipEntry(newPath + "/" + fileList[i] + "/"));
+                        //zip.closeEntry();
```

Comment:
Good catch. This code is directly migrated from the maven-plugins repository with possible enhancements. This line was commented in it and did not try to alter the code a lot due to the fact that the code was working. I ran some tests by closing the entry. We can uncomment the zip.closeEntry() method. Change is done in https://github.com/wso2/carbon-maven-plugins/pull/25